src/plugins/data/common/index_patterns/index_patterns/types.ts

```diff
@@ -33,3 +33,16 @@ export interface TypeMeta {
   aggs?: Record<string, AggregationRestrictions>;
   [key: string]: any;
 }
+
+export interface IndexPatternSpec {
+  id: string;
+  type: string;
```

Comment:
type is always 'index_pattern' so its not needed on the Spec, we can reconstruct index pattern without it

Reply:
interface still needs to be updated

Reply:
objects should not be strings

Reply:
addressed